sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/PartitionReceiverOptions.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Messaging.EventHubs.Core;
+
+namespace Azure.Messaging.EventHubs.Primitives
+{
+    /// <summary>
+    ///   The set of options that can be specified when creating a <see cref="PartitionReceiver" />
+    ///   to configure its behavior.
+    /// </summary>
+    ///
+    internal class PartitionReceiverOptions
+    {
+        /// <summary>The set of options to use for configuring the connection to the Event Hubs service.</summary>
+        private EventHubConnectionOptions _connectionOptions = new EventHubConnectionOptions();
+
+        /// <summary>The set of options to govern retry behavior and try timeouts.</summary>
+        private EventHubsRetryOptions _retryOptions = new EventHubsRetryOptions();
+
+        /// <summary>The amount of time to wait for messages when reading.</summary>
+        private TimeSpan _defaultMaximumReceiveWaitTime = TimeSpan.FromMinutes(1);
+
+        /// <summary>The prefetch count to use for the partition receiver.</summary>
+        private int? _prefetchCount = null;
+
+        /// <summary>
+        ///   The options used for configuring the connection to the Event Hubs service.
+        /// </summary>
+        ///
+        public EventHubConnectionOptions ConnectionOptions
+        {
+            get => _connectionOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(ConnectionOptions));
+                _connectionOptions = value;
+            }
+        }
+
+        /// <summary>
+        ///   The set of options to use for determining whether a failed operation should be retried and,
+        ///   if so, the amount of time to wait between retry attempts.  These options also control the
+        ///   amount of time allowed for reading events and other interactions with the Event Hubs service.
+        /// </summary>
+        ///
+        public EventHubsRetryOptions RetryOptions
+        {
+            get => _retryOptions;
+            set
+            {
+                Argument.AssertNotNull(value, nameof(RetryOptions));
+                _retryOptions = value;
+            }
+        }
+
+        /// <summary>
+        ///   The default amount of time to wait for the requested amount of messages when reading; if this
+        ///   period elapses before the requested amount of messages were available or read, then the set of
+        ///   messages that were read will be returned.
+        /// </summary>
+        ///
+        public TimeSpan DefaultMaximumReceiveWaitTime
```

Comment:
Based on our offline conversation, I'm good with shifting this to nullable, as you recommended.